pkg/stanza/operator/parser/json_array/json_array_parser.go

```diff
@@ -0,0 +1,124 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+package json_array // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/parser/jarray"
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/entry"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/helper"
+
+	"github.com/valyala/fastjson"
+)
+
+const operatorType = "json_array_parser"
+
+func init() {
+	operator.Register(operatorType, func() operator.Builder { return NewConfig() })
```

Comment:
I think unfortunately we need to add these new operators behind a feature gate [(example)](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/82484a272ed2f1d324021c1518b5cbf2c82d9d43/pkg/stanza/fileconsumer/matcher/matcher.go#L119). Otherwise, we're immediately adding them to the filelog receiver, which means no breaking changes to the config. Hopefully we can move the gate forward quickly, but let's at least have one release where it has to be opted into.

Reply:
Done but i wasn't sure about the flag name and usage.. I used `filelog.jsonParserArray` and `filelog.assignKeys`.
let me know if I used it correctly.